Formula/grpc.rb

```diff
@@ -54,18 +53,22 @@ def install
       system "cmake", *args
       system "make", "install"
 
-      args = %w[
-        ../..
-        -DCMAKE_EXE_LINKER_FLAGS=-lgflags
-        -DCMAKE_SHARED_LINKER_FLAGS=-lgflags
-        -DBUILD_SHARED_LIBS=ON
-        -DgRPC_BUILD_TESTS=ON
-        -DgRPC_GFLAGS_PROVIDER=package
-      ] + std_cmake_args
-      system "cmake", *args
-      system "make", "grpc_cli"
-      bin.install "grpc_cli"
-      lib.install Dir["libgrpc++_test_config*.{dylib,so}*"]
+      # grpc_cli does not build correctly with a non-/usr/local prefix.
+      # Reported upstream at https://github.com/grpc/grpc/issues/25176
+      # When removing the `unless` block, make sure to do the same for
+      # the test block.
+      unless Hardware::CPU.arm?
+        args = %W[
+          ../..
+          -DCMAKE_INSTALL_RPATH=#{lib}
+          -DBUILD_SHARED_LIBS=ON
+          -DgRPC_BUILD_TESTS=ON
+        ] + std_cmake_args
+        system "cmake", *args
+        system "make", "grpc_cli"
+        bin.install "grpc_cli"
+        lib.install Dir["libgrpc++_test_config*.{dylib,so}*"]
```

Comment:
Not blocking:

```suggestion
        lib.install Dir[shared_library("libgrpc++_test_config", "*")]
```

Reply:
I'm actually really unhappy about this part of the formula; I'm pretty sure the manual installation is why it doesn't work on M1. I also suspect this wouldn't work on Linux either, because I think it just looks in `/usr/local`.